lib/feature/wallet/staking/view/staking_view.dart

```diff
@@ -0,0 +1,244 @@
+            values: [
+              StakingPageType.stake,
+              StakingPageType.unstake,
+              if (requests != null && requests!.isNotEmpty)
+                StakingPageType.inProgress,
+            ].map(
+              (e) {
+                return CommonTabSwitcherItem(
+                  value: e,
+                  title: e.title,
+                  trailing: e == StakingPageType.inProgress
+                      ? _pendingWithdrawIndicator()
+                      : null,
+                );
+              },
+            ).toList(),
+            currentValue: type,
+          ),
+        ),
+        switch (type) {
+          StakingPageType.stake => _stakeUnstakeBody(),
+          StakingPageType.unstake => _stakeUnstakeBody(),
+          StakingPageType.inProgress => StakingInProgress(
+              requests: requests ?? [],
+              accountKey: accountPublicKey,
+              exchangeRate: exchangeRate ?? 0.0,
+              stakeCurrency: receiveCurrency,
+              attachedFee: attachedAmount,
+              withdrawHours: withdrawTime,
+            ),
+        },
+      ],
+    );
+  }
+
+  Widget _pendingWithdrawIndicator() {
+    return Builder(
+      builder: (context) {
+        final textColor = EverButtonStyleProvider.of(context).contentColor;
+        final colors = context.themeStyle.colors;
+
+        return Container(
+          width: DimensSize.d20,
+          height: DimensSize.d20,
+          alignment: Alignment.center,
+          decoration: BoxDecoration(
+            color: colors.blue,
+            shape: BoxShape.circle,
+          ),
+          child: Text(
+            requests?.length.toString() ?? '0',
+            style: StyleRes.addBold.copyWith(color: textColor),
+          ),
+        );
+      },
+    );
+  }
+
+  Widget _stakeUnstakeBody() {
+    return SeparatedColumn(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      separatorSize: DimensSize.d4,
+      children: [
+        _inputField(),
+        if (currentBalance != null)
+          MoneyWidget(
+            money: currentBalance!,
+            style: MoneyWidgetStyle.secondary,
+          ),
+        const SizedBox(height: DimensSize.d12),
+        _infoField(),
+      ],
+    );
+  }
+
+  Widget _inputField() {
+    return Builder(
+      builder: (context) {
+        final colors = context.themeStyle.colors;
+
+        return CommonInput(
+          titleChild: TonWalletIconWidget(
+            path: imagePath,
+            size: DimensSize.d16,
+          ),
+          titleText: currentBalance != null
+              ? LocaleKeys.tokenAmount.tr(args: [currentBalance!.currency.code])
+              : null,
+          controller: inputController,
+          hintText: '0.0',
+          inputFormatters: [
+            FilteringTextInputFormatter.allow(RegExp(r'[0-9\,\.]')),
+            if (currentBalance != null)
+              CurrencyTextInputFormatter(currentBalance!.currency),
+          ],
```

Comment:
Why do you add additional (regex) formatter and made CurrencyTextInputFormatter optional (depending on currentBalance)?
Also, maybe we should use CurrencyTextInputValidator with min and max values to validate the input?

Reply:
We cannot normally use `CurrencyTextInputValidator` because we will need use Form with key for this action, but Button is out of scope for this widget. But we can use it, if just put the form above

Reply:
Ok, did it, but we cannot use `min` normally because the real min is not just zero

Reply:
-